qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionWithSwitchWhenSectionTest.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.completions;
+
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for Qute completion in expression for #switch and #when section.
+ *
+ */
+public class QuteCompletionInExpressionWithSwitchWhenSectionTest {
+
+	@Test
+	public void propertyPartWhenNoExisting() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is | }\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("ON : MachineStatus", "ON", r(2, 7, 2, 7)), //
+				c("OFF : MachineStatus", "OFF", r(2, 7, 2, 7)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(2, 7, 2, 7)));
+	}
+	
+	@Test
+	public void propertyPartWhenAddtionalSameLine() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is ON |}\r\n" + // <-- completion here
```

Comment:
Are you sure that we  an defines à list of enum  ithiut in key ord? If bo we should have no  œilleton.